README.md

```diff
@@ -1,2 +1,92 @@
+
+One way to think of Item Tasks is that it takes a command line executable and builds a simple Girder web application around that executable. Item Tasks can build a first, and simple UI for an executable, along with a naive approach to running the executable on the server side. 
+
+There are two different ways to self-describe your task for Item Tasks, either using a JSON specification or using standard Slicer Execution Model XML.  Item Tasks aims for compatibility with Slicer XML, but the Slicer XML is designed for a CLI acting on a local filesystem, and so cannot be fully adapted to a CLI operating within the Girder and Girder-Worker environment, i.e. running a task on a remote machine transferring data through Girder.
+
+Specifically the following Slicer XML schema properties are not currently supported in Item Tasks
+
+* geometry
+* point
+* pointfile
+* region
+* table
+* transform
+* any output type that isn't a file
+* the "multiple" modifier that allows the same flag to be provided more than once
+
+If you want to include these input types to your task, the best thing to do is use the conversion tools to translate your XML into the JSON format, and then augment that starting point with the supported JSON spec fields to accomplish what you want.
+
+The conversion script from XML to JSON is located [here](https://github.com/girder/girder/blob/master/plugins/item_tasks/server/cli_parser.py), and this is the code the item_tasks itself uses to convert from Slicer XML to item_tasks internal JSON based representation.
+
+## What Items Tasks is not
+
+Item Tasks is not magic.  It is intended to get simple executables with simple UIs up and running quickly in Girder, which provides great value.
+
+There are many cases where Item Tasks will not suffice.  If the UI is required to be more complex, or customized, or if the algorithm needs to be optimized or scaled for a particular environment or problem, Item Tasks can be a great starting point for further bespoke software development.
+
+## Desired examples
+
+The focus for the examples is on wrapping the executable into the Dockerfile without adding extra complexity.  Having input test data is helpful, but it should be small so as to not pollute git repos.
+
+Having a Dockerfile with a single task and a Dockerfile with multiple tasks are the easiest starting points, but this might be an oversimplification and
+providing the multiple task Dockerfiles may suffice.
+
+* R example, perhaps linear regression
+* Python example, of a wordcount and top n wordcount tasks
+* Python Slicer CLI
+* C++ Slicer CLI
+* GPU/CUDA example
```

Comment:
Sadly the job log won't appear unless you have access, but it's pretty cool - the CNN correctly classified [these two photos](http://10.171.2.112:8080/#folder/593857c0ec929d2e47985b2a) into [these two outputs](http://10.171.2.112:8080/#folder/593857ccec929d2e47985b2c).